tests/messages.cpp

```diff
@@ -459,6 +493,8 @@ TEST(RoomEvents, TextMessage)
         EXPECT_EQ(event.content.msgtype, "m.text");
         EXPECT_EQ(event.content.relates_to.in_reply_to.event_id,
                   "$6GKhAfJOcwNd69lgSizdcTob8z2pWQgBOZPrnsWMA1E");
+
+        EXPECT_EQ(data.dump(), json(event).dump());
```

Comment:
Why are you converting the json to a string, before comparing it?

Reply:
To verify that the conversion from string -> json -> string doesn't lose anything.  We do this same test in a bunch of other places as well.